applications/asset_tracker/src/cloud_codec/cloud_codec.c

```diff
@@ -0,0 +1,385 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+
+#include <stdbool.h>
+#include <string.h>
+#include <zephyr.h>
+#include <zephyr/types.h>
+
+#include "cJSON.h"
+#include "cJSON_os.h"
+#include "cloud_codec.h"
+
+#include <logging/log.h>
+
+LOG_MODULE_REGISTER(cloud_codec, CONFIG_NRF_CLOUD_LOG_LEVEL);
+
+struct cmd {
+	char *name;
+	union {
+		enum cloud_cmd_group group;
+		enum cloud_cmd_recipient recipient;
```

Comment:
The names here can be a bit ambiguous, so I'd say we either change it to something more descriptive or (in any case, actually) place a comment here describing the relationship between the different members of this union.

Reply:
Comments added.